src/lib.rs

```diff
@@ -59,35 +61,35 @@ pub fn use_version(version: &String, compile: bool) -> Result<()> {
     Ok(())
 }
 
-pub fn install_version(version: &String, compile: bool) -> Result<()> {
-    println!("Installing Sui `{version}`");
+pub fn install_version(alias: &String, _compile: bool) -> Result<()> {
+    let version = fetch_version(alias)?;
+
+    println!("Installing Sui `{alias} ({version})`");
 
     #[cfg(all(target_os = "linux", target_arch = "x86_64"))]
-    {
-        if !compile {
-            if let Ok(available_versions) = fetch_versions() {
-                if available_versions.contains(version) {
-                    return download_version(version);
-                }
-            } else {
-                println!("Could not fetch available versions");
-            }
+    if !_compile {
+        let available_versions = fetch_versions()?;
+        if available_versions.contains(&alias) {
+            download_version(&version)?;
+            println!("Downloaded Sui `{alias} ({version})`");
+            return Ok(());
         }
-
-        println!("Could not find version, assuming revision");
-        compile_version(version)?;
     }
-    #[cfg(not(all(target_os = "linux", target_arch = "x86_64")))]
-    compile_version(version)?;
+
+    compile_version(&version)?;
+    println!("Compiled Sui `{alias} ({version})`");
```

Comment:
Flag `compile` could be false yet this command might end up compiling if available versions does not contain the alias

Reply:
Perhaps `fetch_version` instead of `fetch_versions` above?

Reply:
That is the indended behaivour, `fetch_versions` only returns the tags which have available binaries, whereas `fetch_version` transforms the alias string into a commit ID or tag that it can compile (or download) from.

`compile` is only an overwrite in the case that a tag is available to download from but you want to compile. If its not available then we dont have a choice but to compile.